### ISSUE TITLE
XSite NodePort value has no effect

### ISSUE THREAD

**Issue by DanSalt** [infinispan/infinispan-operator#1874]
Based on the documentation, should be able to set a custom NodePort value for Cross Site (especially in cases where firewall rules are being setup to allow traffic between sites)

I have set a section as:
```
    sites:
      local:
        discovery:
          launchGossipRouter: true
          type: gossiprouter
        expose:
          nodePort: 31314
          type: NodePort
```

But the NodePort is assigned randomly to the "site" service for the router.

Looking at the code, it would appear that NodePort is not handled in the function
https://github.com/infinispan/infinispan-operator/blob/8443d1ee715d07e2623d566a9fe699180c68e960/pkg/reconcile/pipeline/infinispan/handler/provision/services.go#L321

**Comment by DanSalt** [comment]
Validated that adding the following at line 307 solves the problem

```
if exposeType == ispnv1.CrossSiteExposeTypeNodePort && exposeConf.NodePort > 0 {
	servicePort.NodePort = exposeConf.NodePort
}
```

**Comment by ryanemerson** [comment]
Thanks for raising the issue @DanSalt. Can you create a PR?

**Comment by ryanemerson** [comment]
PR raised, this will be included in 2.3.7
